src/integration-tests/test_auto_subscriptions.py

```diff
@@ -223,45 +252,145 @@ def test_auto_subscription_with_consumer_subscription(self, cluster: Cluster):
+
         self.consumer.exit_gracefully()
         self.consumer_bar.exit_gracefully()
         self.consumer_baz.exit_gracefully()
-        
+
         cluster.restart_nodes()
-    
+
         self.consumer = self._start_client(
-            proxy, 
-            tc.URI_FANOUT_SC_FOO, 
+            proxy,
+            tc.URI_FANOUT_SC_FOO,
             "consumerFoo",
             subscriptions=[{"correlationId": 1, "expression": "x == 2"}])
-        
+
         self.consumer_bar = self._start_client(
-            proxy, 
-            tc.URI_FANOUT_SC_BAR, 
+            proxy,
+            tc.URI_FANOUT_SC_BAR,
             "consumerBar",
             subscriptions=[{"correlationId": 1, "expression": "x > 2"}])
         self.consumer_baz = self._start_client(proxy, tc.URI_FANOUT_SC_BAZ, "consumerBaz")
-    
+
         self.leader = cluster.last_known_leader
-    
+
         self._verify(tc.DOMAIN_FANOUT_SC, 2)
         self._verify_delivery(self.consumer_bar, 2)
         self._verify_delivery(self.consumer_baz, 2)
-    
+
         self.consumer_bar.confirm(tc.URI_FANOUT_SC_BAR, "*", succeed=True)
         self.consumer_baz.confirm(tc.URI_FANOUT_SC_BAZ, "*", succeed=True)
-        
+
         self._verify(tc.DOMAIN_FANOUT_SC, 0)
 
         assert len(self.consumer_bar.list(block=True)) == 0
-        assert len(self.consumer_baz.list(block=True)) == 0          
-    
\ No newline at end of file
+        assert len(self.consumer_baz.list(block=True)) == 0
+
+    @tweak.domain.subscriptions([{"appId": "", "expression": {"version" : "E_VERSION_1", "text": "x==1"}}])
+    def test_auto_subscription_broadcast(self, cluster: Cluster):
+        """
+        Configure the boadcast queue to evaluate auto subscription negatively.
+        Make sure the queue does not get the message.
+        """
+
+        proxies = cluster.proxy_cycle()
+
+        # 1: Setup producers and consumers
+
+        next(proxies)
+        proxy = next(proxies)
+
+        producer = proxy.create_client("producer")
+        assert (
+            producer.open(tc.URI_BROADCAST, flags=["write", "ack"], block=True)
+            == Client.e_SUCCESS
+        )
+
+        self.consumer = self._start_client(proxy, tc.URI_BROADCAST, "consumer")
+
+        assert (
+            producer.post(
+                tc.URI_BROADCAST,
+                payload=["123"],
+                block=True,
+                wait_ack=True,
+                messageProperties=[{"name": "x", "value": "2", "type": "E_INT"}]
+            )
+            == Client.e_SUCCESS
+        )
+
+        self.leader = cluster.last_known_leader
+
+        self._verify(tc.DOMAIN_BROADCAST, 0)
+
+        assert self.consumer.stop_session(block=True) == Client.e_SUCCESS
+
+        self.consumer.exit_gracefully()
+
+        cluster.restart_nodes()
+
+        self.consumer = self._start_client(proxy, tc.URI_BROADCAST, "consumer")
+
+        self.leader = cluster.last_known_leader
+
+        self._verify(tc.DOMAIN_BROADCAST, 0)
```

Comment:
Since there is no persistence in broadcast mode, the producer would have to post a new message to make sure that the negative auto-subscription is preserved after restart.